tensorflow/lite/kernels/reverse.cc

```diff
@@ -67,40 +61,58 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
 TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
   const TfLiteTensor* input = GetInput(context, node, kInputTensor);
   const TfLiteTensor* axis_tensor = GetInput(context, node, kAxisTensor);
-  int axis = GetTensorData<int32_t>(axis_tensor)[0];
+  int dimension_size = NumDimensions(input);
+  int axes_size = NumElements(axis_tensor);
+  std::vector<bool> axes(dimension_size, false);
+  const int32_t* axis_data = GetTensorData<int32_t>(axis_tensor);
+  int axis = 0;
+  for (int index = 0; index < axes_size; ++index) {
+    axis = axis_data[index];
+    if (axis < 0) {
+      axis += dimension_size;
+    }
+    TF_LITE_ENSURE(context, 0 <= axis && axis < dimension_size);
+    // To ensure no repeating axis
+    TF_LITE_ENSURE(context, axes[axis] == false);
+    axes[axis] = true;
+  }
 
-  TF_LITE_ENSURE(context, axis >= 0 && axis < NumDimensions(input));
   TfLiteTensor* output = GetOutput(context, node, kOutputTensor);
 
   switch (output->type) {
     case kTfLiteFloat32: {
       reference_ops::Reverse<float>(
-          axis, GetTensorShape(input), GetTensorData<float>(input),
-          GetTensorShape(output), GetTensorData<float>(output));
+          axes, NumElements(input), GetTensorShape(input),
```

Comment:
Not very critical but can this additional parameter be avoided and instead do "size_t input_size = NumElements(input);" when entering the Reverse() method in the kernel?